projects/laji/src/app/shared-modules/select/metadata-select-with-subcategories/metadata-select-with-subcategories.component.ts

```diff
@@ -0,0 +1,126 @@
+import { catchError, concatMap, filter, map, switchMap, toArray, tap } from 'rxjs/operators';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, forwardRef, Input, OnChanges, OnDestroy, Output, EventEmitter } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { from, Observable, of, Subscription } from 'rxjs';
+import { WarehouseValueMappingService } from '../../../shared/service/warehouse-value-mapping.service';
+import { Logger } from '../../../shared/logger/logger.service';
+import { CollectionService } from '../../../shared/service/collection.service';
+import { AreaService } from '../../../shared/service/area.service';
+import { SourceService } from '../../../shared/service/source.service';
+import { MetadataService } from '../../../shared/service/metadata.service';
+import { TranslateService } from '@ngx-translate/core';
+import { AdminStatusInfoPipe } from '../admin-status-info.pipe';
+import { Area } from '../../../shared/model/Area';
+import { BaseDataService } from '../../../graph-ql/service/base-data.service';
+import { SelectOptions } from '../select/select.component';
+import { AnnotationService } from '../../document-viewer/service/annotation.service';
+import { MultiLangService } from '../../lang/service/multi-lang.service';
+import { Annotation } from '../../../shared/model/Annotation';
+import { MetadataSelectComponent } from '../metadata-select/metadata-select.component'
+
+
+@Component({
+  selector: 'laji-metadata-select-with-subcategories',
+  templateUrl: './metadata-select-with-subcategories.component.html',
+  styleUrls: ['./metadata-select-with-subcategories.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MetadataSelectWithSubcategoriesComponent extends MetadataSelectComponent implements OnChanges, OnDestroy, ControlValueAccessor {
+  @Input() subCategories = [];
+  @Input() subTitleBase = '';
+  @Input() filtersName = [];
+  @Input() filtersValues = [];
+
+  @Output() update = new EventEmitter<{id: string[] | string, category: string}>();
+
+  lang: string;
+  active = [];
+  selectedTitle = '';
+  _shouldSort = false;
+  _Options:  [{[key: string]: SelectOptions[]}] | Array<any> = [];
```

Comment:
maybe categoryOptions instead.

Typing for is not reflecting the real values most likely should be `{[key: string]: SelectOptions[]}` and then initialised with `{}`. This is used in [here](https://github.com/luomus/laji/pull/9/files#diff-24f96d66f5d15a1249c93013ad5c070d93a6d8d737bd031c20a91aae14b4a03bR111).